crates/fuel-core/src/service/adapters/graphql_api/on_chain.rs

```diff
@@ -103,11 +119,54 @@ impl DatabaseMessages for OnChainIterableKeyValueView {
             .into_boxed()
     }
 
+    fn message_batch<'a>(
+        &'a self,
+        ids: BoxedIter<'a, &'a Nonce>,
+    ) -> BoxedIter<'a, StorageResult<Message>> {
+        <Self as StorageBatchInspect<Messages>>::get_batch(self, ids)
+            .map(|result| result.and_then(|opt| opt.ok_or(not_found!(Messages))))
+            .into_boxed()
+    }
+
     fn message_exists(&self, nonce: &Nonce) -> StorageResult<bool> {
         self.message_exists(nonce)
     }
 }
 
+impl DatabaseCoins for OnChainIterableKeyValueView {
+    fn coin(
+        &self,
+        utxo_id: fuel_core_types::fuel_tx::UtxoId,
+    ) -> StorageResult<fuel_core_types::entities::coins::coin::Coin> {
+        let coin = self
+            .storage::<Coins>()
+            .get(&utxo_id)?
+            .ok_or(not_found!(Coins))?
+            .into_owned();
+
+        Ok(coin.uncompress(utxo_id))
+    }
+
+    fn coins<'a>(
+        &'a self,
+        utxo_ids: BoxedIter<'a, &'a UtxoId>,
+    ) -> BoxedIter<'a, StorageResult<fuel_core_types::entities::coins::coin::Coin>> {
+        let utxo_ids_1: Vec<_> = utxo_ids.collect();
+        let utxo_ids_2 = utxo_ids_1.clone();
```

Comment:
Instead of collecting ids here, you could past `Vec<_>`.

Plus code allows you to do that=) 
<img width="883" alt="image" src="https://github.com/user-attachments/assets/f2dfcf2a-b91d-41f5-81c8-816d845c2836">

Reply:
Right that's better. I've simplified the port in https://github.com/FuelLabs/fuel-core/pull/2396/commits/df33ef2bde75eb7a445232371d1a87926756487d